### ISSUE TITLE
SITL: reimplement the -P command 

### ISSUE THREAD

**Issue by khancyr** [ArduPilot/ardupilot#6114]
#### Issue details
The use of -P is broken as the parameter is erase when the default param file is loaded. 
See @OXINARF comments on for an initial solution https://github.com/ArduPilot/ardupilot/pull/6040

**Comment by ohyaiamhere** [comment]
I would like to take this up, please.

**Comment by Ameya1503** [comment]
Hello ArduPilot Mentors,
I'm Ameya Nanglw, an ECE student with an AI/ML background. I'm actively looking to contribute to open source and am interested in the SITL module.
I'd like to take a look at Issue #6114 to get familiar with the codebase. Is this bug available for me to work on?
I'm also exploring ArduPilot as a potential Google Summer of Code project for next year.
Thanks for your time and guidance!

**Comment by magicrub** [comment]
@ohyaiamhere @Ameya1503 I don't see any pending pull requests about adding this feature so it looks like it's still available to be worked on!

I think the task is something like running sim_vehicle.py with the usual arguments and add -p with a param name and a value, potentially multiple, and make sure they get applied when the autopilot starts up. This would be a shortcut version of using --add-param-file which does the same thing but the params are in a list in a file.

Example:
`sim_vehicle.pr -v Plane -p=SIM_SPEEDUP 2`

**Comment by ohyaiamhere** [comment]
I have created a new pull request #31413 to ameliorate this issue, having modified only the file _sim_vehicle.py_ to do this.

**Comment by hunt0r** [comment]
Should this issue not be closed by #31413 ?

**Comment by khancyr** [comment]
no, the PR only add back -P to sim_vehicle.py but isn't fixing the -P arguement on SITL

**Comment by hunt0r** [comment]
I'm working on this. Can someone specify how SITL's `--param` (aka `-P`) should interact with other options which impact parameters? Specifically, I'm uncertain about `--defaults` and `--wipe`.
(In the absence of a response, I'll do the easiest implementation.)

Related tangents:
- Looks like `--model` is designed to overwrite everything, so I assume it should also overwrite `-P` stuff.
- The changes provided by `-P` will be saved to the param file, and thus by-default become loaded in the next run.

**Comment by hunt0r** [comment]
It's not clear to me if this issue desires changes to how `sim_vehicle.py` handles `-P` after SITL supports it.
The best path forward seems to be a PR to fix SITL's `-P`. Any desired changes to `sim_vehicle` can be done in a separate PR after.

**Comment by peterbarker** [comment]
@hunt0r just deleted the thing - we decided to do things this way on DevCall.

Closing this one now
